File: src/main/java/rr/language/RRIconProvider.java
package rr.language;

import com.intellij.ide.IconProvider;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class RRIconProvider extends IconProvider {

    public Icon getIcon(@NotNull PsiElement psiElement, int flags) {
        PsiFile containingFile = psiElement.getContainingFile();

        if (containingFile == null) return null;

        if (!RRUtil.isScript(containingFile)) return null;

        return RRIcons.SCRIPT_FILE;
    }
}

File: src/main/java/rr/language/RRIconProvider.kt
package rr.language

import com.intellij.ide.IconProvider
import com.intellij.psi.PsiElement
import javax.swing.Icon

class RRIconProvider : IconProvider() {
    override fun getIcon(psiElement: PsiElement, flags: Int): Icon? {
        val containingFile = psiElement.containingFile ?: return null
        return if (!RRUtil.isScript(containingFile)) null else RRIcons.SCRIPT_FILE
    }
}